ccpp/data/CCPP_typedefs.F90

```diff
@@ -1141,8 +1141,12 @@ subroutine gfs_interstitial_setup_tracers(Interstitial, Model)
     if (Model%ntke > 0) Interstitial%ntkev = Interstitial%nvdiff
 
     if (Model%ntiw > 0) then
-      if (Model%ntclamt > 0) then
+      if (Model%ntclamt > 0 .and. Model%ntsigma == 0) then
         Interstitial%nn = Model%ntrac - 2
+      elseif (Model%ntclamt == 0 .and. Model%ntsigma > 0) then
+        Interstitial%nn = Model%ntrac - 2
+      elseif  (Model%ntclamt > 0 .and. Model%ntsigma > 0) then
+        Interstitial%nn = Model%ntrac - 3
       else
```

Comment:
@JongilHan66 @ChunxiZhang-NOAA this statement had me a bit puzzled, I'm not sure this is the right solution. Essentially I do not want to vertically diffuse updraft area fraction in PBL or convection, and I am aiming at treating ntsigma as ntclamt in this regard. So if either is present it should be reduced by 2, if both are present it should be reduced by 3. Maybe there is a better way?

Reply:
@ChunxiZhang-NOAA yes, I do. And I'd made sure that is done correctly. I'm speaking of the handling of the vertical transport in physics. This hard coded segment is perhaps a bit dangerous as there are many combinations of tracers in the field tables. Ideally perhaps a flag could go with each tracer whether to diffuse or advect or both, but right now I'm just adding to this existing code.

Reply:
@lisa-bengtsson I was thinking if updraft area fraction was not advect in dynamics, it didn't need to be defined as tracer, which could make things simple. Tracers are treated in a complicated way in CCPP.  Maybe we can find a better way in the future. Back to the current code, I believe the index number for missing one is -99.  It would be safer to use <=0.

Reply:
@ChunxiZhang-NOAA ok, yes that is a good idea, I will change to <=. 

Yes the discussion on tracer treatment within the physics is beyond the scope of this PR, but I hope we can avoid these type of if-statements in the future.  Thanks. I will address this and your other comments today. Thanks for having a careful look.

Reply:
@lisa-bengtsson I don't get it on why you advect the sub-grid updraft fraction horizontally and vertically by the mean wind. Does it physically make sense? As I remember, the original equation for the prognostic updraft equation does not have advection term.

Reply:
@JongilHan66 in Gerard et al. 2009 the area fraction is advected, also in the 3MT scheme. There is no advection term, but area fraction from the previous time-step is advected before going into termA. This is also what is done in our paper.

Reply:
@lisa-bengtsson it's hard to understand, considering the cumulus updraft is a sub-grid scale process. How can the sub-grid fraction go across the grid box? Also, the source term is moisture convergence, which is a result of moisture advection.

Reply:
@JongilHan66 I suppose it would be a similar argument for advecting turbulent kinetic energy? It is a sub-grid scale process with local buoyancy/shear production and dissipation. If a larger area fraction represents "stronger" or "more organized" convection, I think it would make sense to advect this with the flow as e.g. meso-scale convective systems could be moved with the flow.